src/controllers/message/MessageController.ts

```diff
@@ -75,284 +49,24 @@ export class MessageController {
   })
   public async sendMessage(@Body() message: IBaseMessage): Promise<{ id: string }> {
     try {
-      const agent = await this.agentService.getAgent()
-
-      let messageId: string | undefined
-      const messageType = message.type
-      this.logger.debug!(`Message submitted. ${JSON.stringify(message)}`)
-
-      const connection = await agent.connections.findById(message.connectionId)
+      let connection: ConnectionRecord | null = null
+      connection = await this.checkForDuplicateId(message)
```

Comment:
IMHO this method should return undefined, throwing an Error in case of duplicate like it does right now), and the line `const connection = await agent.connections.findById(message.connectionId)` remain here.

Reply:
fixed